src/Sprites/IndexedBmpWriter.cpp

```diff
@@ -71,27 +47,33 @@ void IndexedBmpWriter::WriteHeaders(Stream::SeekableWriter& seekableWriter, uint
 	BmpHeader bmpHeader = BmpHeader::Create(static_cast<uint32_t>(fileSize), static_cast<uint32_t>(pixelOffset));
 	ImageHeader imageHeader = ImageHeader::Create(width, height, bitCount);
 
-	seekableWriter.Write(bmpHeader.type);
-	seekableWriter.Write(bmpHeader.size);
-	seekableWriter.Write(bmpHeader.reserved1);
-	seekableWriter.Write(bmpHeader.reserved2);
-	seekableWriter.Write(bmpHeader.pixelOffset);
-
+	seekableWriter.Write(bmpHeader);
 	seekableWriter.Write(imageHeader);
 }
 
-int32_t IndexedBmpWriter::CalculateScanLineSize(uint16_t bitCount, int32_t width)
+void IndexedBmpWriter::WritePixels(Stream::SeekableWriter& seekableWriter, uint16_t bitCount, int32_t width, int32_t height, const std::vector<uint8_t>& pixels)
+{
+	// If height > 0, top line of pixels is bottom line in file. 
+	// If height < 0, top line of pixels is top line in file
+	if (height < 0) {
+		WritePixelsTopDown(seekableWriter, bitCount, width, height, pixels);
+	}
+	else {
+		WritePixelsBottomUp(seekableWriter, bitCount, width, height, pixels);
+	}
```

Comment:
Hmm. This could be nice, though if it complicates things, I don't know that we really need to support writing both formats. I assume bottom up is pretty rare and unexpected these days. It might make more sense to only support both formats for loading, and convert to a standard format which is always used for writing.

Reply:
Well, it is already written and working. MSPaint appears to default to saving indexed bitmaps in bottom up format, so I suspect we may encounter them. It may be more difficult to test the BmpReader for bottom up if we cannot produce them. I'd like to leave it in for now. If it becomes a burden to maintain, we could revisit.